pkg/cache/context.go

```diff
@@ -552,6 +552,16 @@ func (ctx *Context) RemoveApplication(appID string) error {
 	}
 }
 
+func (ctx *Context) RemoveApplicationInternal(appID string) error {
```

Comment:
can you add a UT to cover this function?

Reply:
OK, done